src/main/java/org/spongepowered/api/entity/ai/task/builtin/creature/location/MoveToBlockAITask.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.ai.task.builtin.creature.location;
+
+import com.flowpowered.math.vector.Vector3i;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.entity.ai.task.AITask;
+import org.spongepowered.api.entity.ai.task.AITaskBuilder;
+import org.spongepowered.api.entity.living.Creature;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.util.function.Predicate;
+
+public interface MoveToBlockAITask extends AITask<Creature> {
+
+    static MoveToBlockAITask.Builder builder() {
+        return Sponge.getRegistry().createBuilder(MoveToBlockAITask.Builder.class);
+    }
+
+    double getSpeed();
+
+    int getSearchRange();
+
+    Vector3i getDestination();
+
+    MoveToBlockAITask setDestination(Vector3i destination);
+
+    interface Builder extends AITaskBuilder<Creature, MoveToBlockAITask, MoveToBlockAITask.Builder> {
+
+        MoveToBlockAITask.Builder speed(double speed);
+
+        MoveToBlockAITask.Builder searchRange(int searchRange);
+
+        MoveToBlockAITask.Builder destinationPredicate(Predicate<Location<World>> destinationPredicate);
```

Comment:
I didn't include that in the builder as it doesn't really make sense, it's something that's set later to override the current state.